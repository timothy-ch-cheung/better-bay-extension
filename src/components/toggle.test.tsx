import { describe, expect, test } from "@jest/globals"
import { render } from "@testing-library/react"

import Toggle from "./toggle"

describe("Toggle", () => {
  test("Snapshot", () => {
    const tree = render(
      <Toggle label="Test Toggle" enabled={true} onClick={jest.fn()} />
    )
    expect(tree).toMatchSnapshot()
  })
})
