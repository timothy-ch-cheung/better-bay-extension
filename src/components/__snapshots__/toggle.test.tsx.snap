// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="inline-flex relative items-center cursor-pointer"
        data-test="test-toggle-toggle"
      >
        <input
          checked=""
          class="sr-only peer"
          data-test="test-toggle-toggle"
          type="checkbox"
          value=""
        />
        <div
          class="w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-[2px] after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600"
        />
        <span
          class="ml-3 text-sm font-medium text-gray-900 dark:text-gray-300"
        >
          Test Toggle
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="inline-flex relative items-center cursor-pointer"
      data-test="test-toggle-toggle"
    >
      <input
        checked=""
        class="sr-only peer"
        data-test="test-toggle-toggle"
        type="checkbox"
        value=""
      />
      <div
        class="w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-[2px] after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600"
      />
      <span
        class="ml-3 text-sm font-medium text-gray-900 dark:text-gray-300"
      >
        Test Toggle
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
